odoo/addons/base/models/ir_async.py

```diff
@@ -0,0 +1,299 @@
+        to notify the job started. A second message is sent at the end
+        containing either the model method returned value, either an
+        error along with the exception and the complete traceback.
+        """
+        for json_field in ('res_ids', 'args', 'kwargs', 'env_context'):
+            job[json_field] = json.loads(job[json_field])
+        job['env_context']['async_traceback'] = job['traceback']
+
+        env = api.Environment(job_cr, job['env_uid'], job['env_context'])
+        ir_async = env['ir.async']
+
+        ir_async._pre_process(job)
+
+        try:
+            _logger.info('Calling "%s" on "%s"', job['method'], job['res_model'])
+            records = env[job['res_model']].browse(job['res_ids'])
+            result = getattr(records, job['method'])(*job['args'], **job['kwargs'])
+            json.dumps(result)  # ensure result is serializable
+            records.flush()
+        except Exception as exc:
+            payload = ir_async._handle_exception(exc)
+        else:
+            _logger.debug("Done, result is %s", result)
+            payload = ir_async._json_response(result)
+
+        ir_async._post_process(job, payload)
+
+    @api.model
+    def _merge_traceback(self, tb):
+        """ Merge the given traceback with all previous level of execution """
+        old_tb = self.env.context.get('async_traceback', '').splitlines(keepends=True)
+        if old_tb:
+            def fnc_process_job_not_reached(line):
+                return not re.search(f'in {self._process_job.__name__}\\W', line)
+            old_tb.append("Traceback of async job (most recent call last):\n")
+            old_tb.extend(itertools.dropwhile(fnc_process_job_not_reached, tb))
+            tb = old_tb
+        return "".join(tb)
+
+    @api.model
+    def _pre_process(self, job):
+        pass
+
+    @api.model
+    def _post_process(self, job, payload):
+        pass
+
+    @api.model
+    def _handle_exception(self, exception):
+        header, *tb = traceback.format_exc().splitlines(keepends=True)
+        exc_tb = header + self._merge_traceback(tb)
+
+        if isinstance(exception, odoo.exceptions.UserError):
+            _logger.warning(exception)
+        else:
+            _logger.error("Failed to process async job\n%s", exc_tb)
+
+        error = {
+            'code': 200,
+            'message': "Odoo Server Error",
+            'data': serialize_exception(exception),
+        }
+        error['data']['debug'] = exc_tb
+        return self._json_response(error=error)
+
+    @api.model
+    def _json_response(self, result=None, error=None):
+        response = {}
+        if result is not None:
+            response['result'] = result
+        if error is not None:
+            response['error'] = error
+        return response
+
+
+class ir_async_export(models.TransientModel):
+    _name = 'ir.async.export'
+    _description = "Model helper for ExportFormat"
```

Comment:
That really absolutely doesn't belong in base.

Reply:
Where then ?

Reply:
Given it's a helper for something which lives in `web` and it literally depends on (imports) `web`, `web` seems like a good idea. In fact I'm not quite sure why this model even exists, why isn't this  just straight part of ExportFormat or its descendants?